snaps/how-to/use-keyring-api/create-account-snap.md

```diff
@@ -0,0 +1,225 @@
+- [Keyring API reference](../../reference/keyring-api/index.md)
+:::
+
+## Prerequisites
+
+- Set up a Snap.
+  See the [Snaps quickstart](../../get-started/quickstart.md) and [how to develop a Snap](../develop-a-snap.md).
+- Read the [account management Snap security guidelines](security.md).
+
+## Steps
+
+### 1. Install the Keyring API
+
+Install `@metamask/keyring-api` in your project directory using Yarn or npm:
+
+```bash
+yarn add @metamask/keyring-api
+```
+
+or
+
+```bash
+npm install @metamask/keyring-api
+```
+
+### 2. Add permissions
+
+Specify the following [permissions](../request-permissions.md) in your Snap manifest file:
+
+```json title="snap.manifest.json"
+"initialPermissions": {
+  "endowment:keyring": {
+    "allowedOrigins": [
+      "https://<dapp domain>"
+    ]
+  },
+  "endowment:rpc": {
+    "dapps": true
+  },
+  "snap_manageAccounts": {},
+  "snap_manageState": {}
+},
+```
+
+Add a list of URLs of dapps allowed to call Keyring API methods on your Snap using the
+[`endowment:keyring`](../../reference/permissions.md#endowmentkeyring) permission.
+
+### 3. Implement the Keyring API
+
+Implement the [required Keyring API methods](security.md#limit-the-methods-exposed-to-dapps) in your Snap:
+
+```typescript
+class MySnapKeyring implements Keyring {
+  // Implement the required methods here...
+}
+```
+
+### 4. Handle requests submitted by MetaMask
+
+MetaMask will submit Ethereum sign requests from dapps using the
+[`submitRequest`](../../reference/keyring-api/type-aliases/Keyring.md#submitrequest) method of the
+Keyring API.
+See the [supported signing methods](../../concepts/keyring-api.md#supported-signing-methods).
+
+The following is an example of a `personal_sign` request:
+
+```json
+{
+  "id": "d6e23af6-4bea-48dd-aeb0-7d3c30ea67f9",
+  "scope": "",
+  "account": "69438371-bef3-4957-9f91-c3f22c1d75f3",
+  "request": {
+    "method": "personal_sign",
+    "params": [
+      "0x4578616d706c652060706572736f6e616c5f7369676e60206d657373616765",
+      "0x5874174dcf1ab6F7Efd8496f4f09404CD1c5bA84"
+    ]
+  }
+}
+```
+
+The request includes:
+
+- `id` - The unique identifier for the request.
+- `scope` - The CAIP-2 chain ID of the selected chain.
+  Currently, this property is always an empty string.
+  Your Snap should use the chain ID present in the request object instead.
+- `account` - The ID of the account that should handle the request.
+- `request` - The request object.
+
+Your Snap must respond with either a synchronous result:
+
+```typescript
+return { pending: false, result };
+```
+
+Or an asynchronous result:
+
+```typescript
+return { pending: true, redirect: { message, url } };
+```
+
+The redirect message and URL is displayed to the user to inform them on how to continue the transaction flow.
+
+### 5. Notify MetaMask about events
+
+Notify MetaMask when the following events take place, using the `emitSnapKeyringEvent()` helper function.
+
+1. An account is created:
```

Comment:
The topic is step #5. Do you want to make the steps within this topic alphabetic...a, b, c...? Also, it seems that the numbered items here are results and not steps. Do you want to use bullets instead?